docs/python.md

```diff
@@ -103,6 +103,19 @@ The `Input()` function takes these keyword arguments:
 
 Each parameter of the `predict()` method must be annotated with a type like `str`, `int`, `float`, `bool`, etc. See [Input and output types](#input-and-output-types) for the full list of supported types.
 
+Using the `Input` function is recommended because it provides better documentation and validation constraints to the users of your model, but it is not strictly required. You can also specify default values for your parameters using plain Python, or omit default assignment entirely:
+
+```py
+class Predictor(BasePredictor):
+    def predict(self, 
+        prompt: str = "default prompt", # this is valid
+        iterations: int                 # also valid
+    ) -> str:
+        for x in (iterations or 1:)
+            return "hello, " + prompt
```

Comment:
Should this be `yield`, for the spirit of the example? As is, the number of iterations won't matter.

Reply:
Removed the function body for clarity ad75f4d6d0e5751254c05b0ea1f192b768fa2fa9